src/app/models/treeoperations.cpp

```diff
@@ -157,3 +157,19 @@ void TreeOperations::notifyDbWasUnloaded(int dbIndex)
 {
     emit closeDbKeys(m_connection, dbIndex);
 }
+
+void TreeOperations::deleteDbKey(int dbIndex, ConnectionsTree::KeyItem& key)
```

Comment:
As second arg you should add callback `std::function<void(const QString&)> callback`